ibm/data_source_ibm_common_schema.go

```diff
@@ -0,0 +1,49 @@
+package ibm
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func filterIBMDataSources(set *schema.Set, data *[]map[string]interface{}) *[]map[string]interface{} {
+	var filteredVals []map[string]interface{}
+	for _, v := range set.List() {
+		m := v.(map[string]interface{})
+		key := m["name"].(string)
```

Comment:
Do type casting only if value exists..if not it might result in nil pointer exception..
check if m["name"] exists or not..
```
var key string
if k,ok := m["name"];ok{
  key =k.(string)
}
```

same applies for values..